COMPILING.md

```diff
@@ -1,19 +1,125 @@
+     [the github release page](https://github.com/lexxmark/winflexbison/releases/)
+     or through
+     [the chocolatey package manager.](https://community.chocolatey.org/packages/winflexbison3)
+     Installing strawberryperl is advised if you want to run tests on Windows.
+   - Use of CMake has not been tested on Solaris or FreeBSD. However, it should
+     be possible to install CMake from the system package manager or the
+     [official download page](https://cmake.org/download) on those systems.
+     The dependencies (as listed in the relevant sections above) will also be
+     required, and should be installed using the suggested methods.
+2. Navigate to the *top level* folder of the project. This is different from
+   the Makefile build, which requires you to navigate to the `src` directory
+   first.
+3. Update git submodules:
+   ```
+   git submodule update --init
+   ```
+4. Generate build files with CMake:
+   ```
+   cmake -S . -Bbuild
+   ```
+   This command tells CMake to use the configuration in the current directory,
+   and to generate build files into the `build` directory.  This is the point
+   to specify custom build settings, such as compilers and build back-ends. You
+   can use clang (for example) by adding the argument
+   `-DCMAKE_CXX_COMPILER=clang++` to the command line. You can also tell CMake
+   to generate IDE projects by supplying the `-G` flag.  Run `cmake -G` for a
+   comprehensive list of supported back-ends.
+
+   On macOS >10.14, the build will fail unless you explicitly specify
+   the full path to the compiler. This issue is being tracked
+   [here](https://github.com/diffblue/cbmc/issues/4956). The invocation thus
+   looks like this:
+   ```
+   cmake -S. -Bbuild -DCMAKE_C_COMPILER=/usr/bin/clang
+   ```
+
+   Generally it is not necessary to manually specify individual compiler or
+   linker flags, as CMake defines a number of "build modes" including Debug and
+   Release modes. To build in a particular mode, add the flag
+   `-DCMAKE_BUILD_TYPE=Debug` (or `RelWithDebInfo`) to the initial invocation.
+   The default is to perform an optimized build via the `Release` configuration.
+
+   If you *do* need to manually add flags, use `-DCMAKE_CXX_FLAGS=...` and
+   `-DCMAKE_EXE_LINKER_FLAGS=...`. This is useful for enabling clang's
+   sanitizers.
+
+   Finally, to enable building universal binaries on macOS, you can pass the
+   flag `-DCMAKE_OSX_ARCHITECTURES=i386;x86_64`. If you don't supply this flag,
+   the build will just be for the architecture of your machine.
+5. Run the build:
+   ```
+   cmake --build build
+   ```
+   This command tells CMake to invoke the correct tool to run the build in the
+   `build` directory. You can also use the build back-end directly by invoking
+   `make`, `ninja`, or opening the generated IDE project as appropriate. The
+   complete set of built binaries can be found in `./build/bin` once the build
+   is complete.
```

Comment:
Perhaps add a comment about doing parallel builds in here? Most people will have multiple cores available these days, and the build process is a lot faster when using more than one core...